### ISSUE TITLE
tab_new fails when in a directory with spaces in it

### ISSUE THREAD

**Issue by rightaway** [ranger/ranger#1315]
If the directory you're currently in has spaces in the path, then opening a new tab fails when you've configured ranger to open the new tab in the same directory as the current tab.

**Comment by rightaway** [comment]
This is the relevant part in the rc `map tn tab_new %d`, it fails when I use the shortcut.

**Comment by toonn** [comment]
I can confirm this is a bug. The path is not escaped and the spaces make it parse as separate arguments.

In this case there's a workaround though. Drop the `%d`, `tab_new` without arguments opens a new tab in the same directory. It's also already mapped to both `<C-n>` and `gn` btw.

**Comment by j-degreef** [comment]
Ok, that's the bug I've noticed (see [#1392](https://github.com/ranger/ranger/issues/1392)).

The work around is ok to open another tab in the current directory but not to open a tab and passing a directory to it (tab_new /home/me/directory with spaces) which would be very useful to map new_tab to directories that we use regularly.

**Comment by toonn** [comment]
#1392 wasn't a bug as far as I'm concerned. So unless you can provide a reproducible scenario, stop spreading FUD ; )

**Comment by j-degreef** [comment]
You misunderstand me : I agree #1392 is not a bug.
I've put this number here because I was talking, in that post, about a bug I had found with tab_new. That bug I was talking about is the bug described here.
See this line in #1392 : "Btw, I think I have discovered 2 other problems with "dir with spaces" (with tab_new and shell cp) but this time I'll double check before creating an issue.",
And to be complete, there's no bug with "shell cp".
I hope this clarify this matter.

**Comment by toonn** [comment]
Just to be clear, I was joking about the FUD ; )

**Comment by j-degreef** [comment]
:-)

**Comment by palikar** [comment]
@toonn, I am currently looking into this issue and the bug with the arguments. Would you be interested in a change that would allow to use every command like
```
map tn tab_new "%d"
``` 
The quotes would cause ranger to treat everything inside them as a single argument.

**Comment by toonn** [comment]
@palikar, no. The responsibility of escaping is on the macro substituting code, not on the mapping writer. It wouldn't be possible to fix it like this if the macro was `%s` for example. Each of the items would have to be escaped separately, not all of them as a whole and you don't have access to the individual items when composing a map.

**Comment by palikar** [comment]
@toonn Ok, I see what you mean. If for example you visually select a directory(or a file) with spaces in it and use it somehow through ` %s`, the called function will actually see two separate arguments. If you, however, visually select two directories, one with spaces and what without, and you call `:fun "%s"` will again get  a single argument - a string with the names of the directories. If the function is called as ":fun %s", it will see three separate arguments. Again - not what is to be expected.

This suggests, however, that ranger need escaping login both on the side of commands *and* on the macros expansion code. Currently it is not possible for a command to accept an argument with spaces in it (as far as I see\understand from the code). OK, in this case, would you be interested in a fix on the both ends?

**Comment by toonn** [comment]
You're right. `tab_new` doesn't deal with quoted arguments so yes we'll need at least two fixes.

**Comment by manujchandra** [comment]
The following is not working:

`map tra tab_new ~/Productivity_System/01 TASK CAPTURE BIN/00 COPY TO HDD/RELAX PASSIVE ASSORTED`

**Comment by toonn** [comment]
Yes, tab_new expects multiple arguments separated by spaces. That's why we need to be able to escape arguments. It's really quite broken currently tbh.
The only way to work around it currently is to use `:eval self.fm.tab_new(path="My path/with spaces")` to call the function behind the command without parsing interfering.

**Comment by Grawp** [comment]
@toonn Thank you for the workaround. I needed to bind a shortcut for opening selected directory in a new tab. This done it:
```
map O eval self.fm.tab_new(path=self.fm.thisfile.relative_path)
```
This of course work not only with directories with spaces in their filename but also with `"` and `'` or any other character :)
